src/Chapter4.hs

```diff
@@ -489,6 +495,26 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+instance Applicative List where
+    pure :: a -> List a
+    pure a = Cons a Empty
+
+    (<*>) :: List (a -> b) -> List a -> List b
+    Empty <*> _ = Empty
+    _ <*> Empty = Empty
+    fs <*> vs = concatL (fmap (\v -> fmap (\f -> f v ) fs ) vs)
```

Comment:
Nice usage of nested `fmap`s! Since `fmap` and `concatL` handle empty lists quite well, I think you don't need to pattern-match on `Empty` lists separately :slightly_smiling_face:

Reply:
Thanks, I added tests for that cases as well